packages/core/src/extensions/DraggableBlocks/DraggableBlocksPlugin.ts

```diff
@@ -341,33 +421,43 @@ export const createDraggableBlocksPlugin = (editor: Editor) => {
             return true;
           }
 
-          const rect = absoluteRect(blockContent);
-          const win = block.node.ownerDocument.defaultView!;
-          const dropElementRect = dropElement.getBoundingClientRect();
-          const left =
-            (horizontalPosAnchoredAtRoot ? getHorizontalAnchor() : rect.left) -
-            WIDTH +
-            win.pageXOffset;
-          rect.top +=
-            rect.height / 2 - dropElementRect.height / 2 + win.pageYOffset;
-
-          dropElement.style.left = left + "px";
-          dropElement.style.top = rect.top + "px";
-
-          // MantineProvider has been commented out because I removed
-          // BlockNoteTheme. I know this will be part of the DraggableBlocks rewrite anyway
-          dropElementRoot.render(
-            // <MantineProvider theme={BlockNoteTheme}>
-            <DragHandle
-              key={block.id + ""}
-              editor={editor}
-              coords={coords}
-              onShow={onShow}
-              onHide={onHide}
-              onAddClicked={onAddClicked}
-            />
-            // </MantineProvider>
-          );
+          // Gets bounding box of relevant block.
+          const blockBoundingBox = blockContent.getBoundingClientRect();
+          blockBoundingBox.x = horizontalPosAnchoredAtRoot
+            ? getHorizontalAnchor()
+            : blockBoundingBox.left;
+
+          // Updates element params.
+          addBlockButtonParams.addBlock = () =>
+            addBlock({
```

Comment:
I don't think the functions addblock / deleteblock should depend on coords. We already know which block it is right? Can't we pass the block or blockid?

Reply:
Having the block's underlying PM node itself doesn't really help here since the actual editor functions to add and delete blocks (`BNCreateBlock()` and `BNDeleteBlock()`) take in a position inside a block as an argument. Since PM doesn't treat blocks as unique objects, even if we had the PM node, we would have to find it in the document, which wouldn't be any more efficient. So `view.posAtCoords()` turns out to be the easiest way of getting the correct position.